pandas/io/formats/style_render.py

```diff
@@ -809,7 +810,9 @@ def pred(part) -> bool:
             # slice(a, b, c)
             slice_ = [slice_]  # to tuplize later
     else:
-        slice_ = [part if pred(part) else [part] for part in slice_]
+        # error: Item "slice" of "Union[slice, Sequence[Any]]" has no attribute
+        # "__iter__" (not iterable) -> is specifically list_like in conditional
+        slice_ = [p if pred(p) else [p] for p in slice_]  # type: ignore[union-attr]
```

Comment:
if you return this directly it should work (as you are re-assigning to a different type)

Reply:
it wasnt the reassignment it was the fact that slice_ is not always considered to be iterable inside the loop comprehension. Albeit in this clause we have already checked that it is `list_like` so is iterable here...